### ISSUE TITLE
[Feature request] give a file as input to download several profiles

### ISSUE THREAD

**Issue by althonos** [althonos/InstaLooter#17]
Got a mail request requesting to be able to provide instaLooter with a file containing a list of profiles/directories to download files from/into.

Suggested format was JSON, but I think YAML would be easier to edit from an user POV. Easiest would be a YAML list following the exact syntax you'd use if you ran individual jobs. For instance:
```yaml
- user-we-dont-want-vids-from dir1
- user-we-want-vids-from dir2 --get-videos
```

Even better, it could be possible to give parameters you'd like to use everywhere (so you'd not have to give them back at each run):
```yaml
params: 
  -c MYLOGIN --get-videos --add-metadata
jobs:
- user1 dir1
- user2 dir2 -N
```

**Comment by mrgloom** [comment]
I was using this bash one-liner to download photos from users list `cat user_list.txt | xargs -n 1 --max-procs=1 -I {} instaLooter -N {} images/{}`, where `user_list.txt` is just .txt file with one user name per line.

p.s. with `--max-procs=2` it works faster, but seems it breaks progress bar, also I tried to use `-j 32` parameter of your app, but it give me no speedup, so I wonder how to set optimal number of jobs?

**Comment by althonos** [comment]
Nice ! I am no bash ninja so I'd like a python solution that works like that, but you got the gist of it. If you don't want progress bar because it looks shitty with multiprocessing, you can use the `-q` option to tell instaLooter to shut up.

The way instaLooter works makes it slightly unefficient to use multiprocessing on one profile, though. Since it doesn't use the Instagram API, you'd have to do the following:
```
load page 1 ==> download media 1 ==> download media 2 ==> ... ==> download media 12 ==> load page 2 ==> download media 13 ==> ... ==> ...
```

I enhanced that a bit, so what actually happens is:
```raw
load page 1 =====================> load page 2====================> ...
     \--> register files 1-12            \--> register files 13-24
           with background workers             with background workers
```

The `-j` argument can only speed up the download of files, but doesn't speed up loading all profile pages.


With what I'd like to do though, you'd be able to give different parameters for each username, which would make it easier. Running all jobs in parallel would be easy too.

**Comment by althonos** [comment]
## SO...
... I've implemented a batch list mecanism using the Python *config* format. Feel free to try it after installing from the master branch.

#### Command
`ìnstaLooter batch <batch_file>`

#### File format (example)
```ini
[some private users]   # any ID is fine here
username = foo   # the targets in this job will be downloaded after logging in
password = bar

get-videos = True
add-metadata = True

users = 
    someuser: /some/directory
    anotheruser: /some/other/directory

[funny videos]
username = spam  # no password given, so it will be asked in the shell

videos-only = true
num-to-dl = 10
new = true

hashtags =
    funny: ../local/path/to/my/funny/videos

[video games]
# no user, so downloads will be done without logging in

users =
    squareenix: /my/videos/games/directory
hashtags =
    finalfantasy: /my/videos/games/directory
    dragonquest: /my/videos/games/directory
```

**Comment by althonos** [comment]
Available in `v0.12.0`. Refer to http://instalooter.readthedocs.io/en/latest/batch.html for more details about the format.
